jazz_assets/components/validation.js

```diff
@@ -0,0 +1,311 @@
+            if (assets_data[field]) {
+                if (!validateDataTypes(field, assets_data[field])) {
+                    invalid_fields.push(field);
+                }
+            }
+        })
+
+        if (invalid_fields.length > 0) {
+            var message = "The following field's value/type is not valid - " + invalid_fields.join(", ");
+            reject({
+                "result": "inputError",
+                "message": message
+            });
+        } else {
+            resolve({
+                "result": "success",
+                "input": "Input value is valid"
+            });
+        }
+    });
+};
+
+function validateDataTypes(field, prop_value) {
+    logger.debug("Inside validateDataTypes");
+    var fields_type = global.global_config.FIELD_DATA_TYPES;
+    var field_status = false;
+    Object.keys(fields_type).map((type) => {
+        if (field === type) {
+            if (fields_type[type] === 'String') {
+                if (prop_value && (typeof prop_value === 'string' || prop_value instanceof String)) {
+                    field_status = true;
+                }
+            } else if (fields_type[type] === 'Array') {
+                if (prop_value && prop_value.constructor === Array) {
+                    field_status = true;
+                }
+            }
+        }
+    });
+    return field_status;
+};
+
+function validateEnumValues(assets_data) {
+    logger.debug("Inside validateEnumValues");
+    return new Promise((resolve, reject) => {
+        var invalid_fields = [];
+        Object.keys(assets_data).map((field) => {
+            if (assets_data[field]) {
+                var value = assets_data[field];
+                switch (field) {
+                    case 'status':
+                        if (global.global_config.ASSET_STATUS.indexOf(value) === -1) {
+                            invalid_fields.push(field);
+                        }
+                        break;
+                    case 'type':
+                        if (global.global_config.ASSET_TYPES.indexOf(value) === -1) {
+                            invalid_fields.push(field);
+                        }
+                        break;
+                }
+            }
+        });
+
+        if (invalid_fields.length > 0) {
+            var message = "The following field's value are not valid - " + invalid_fields.join(", ");
+            reject({
+                "result": "inputError",
+                "message": message
+            });
+        } else {
+            resolve({
+                "result": "success",
+                "input": "Input value is valid"
+            });
+        }
+    });
+};
+
+function validateEditableFields(update_data, editableFields) {
+    logger.debug("Inside validateEditableFields");
+    var invalid_fields = _.difference(_.keys(update_data), _.values(editableFields));
+    invalid_fields.map((value) => {
+        delete update_data[value];
```

Comment:
are you sure you want to mutate the original object using map or do you want to use something like filter on the original object and extract what is needed into a new one and use it. Generally its not a good idea to mutate objects like that (specially using a map, its not functional).

Reply:
Using filter function to remove invalid fields.